server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1633,6 +1637,13 @@ private SearchSourceBuilder parseXContent(
                     }
                     searchUsage.trackSectionUsage(KNN_FIELD.getPreferredName());
                 } else if (SUB_SEARCHES_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    if (parser.getRestApiVersion().matches(RestApiVersion.onOrAfter(RestApiVersion.V_8))) {
+                        deprecationLogger.warn(
+                            DeprecationCategory.API,
+                            "sub_searches_api",
+                            "Using [sub_searches] is deprecated and will be removed in a future release"
+                        );
+                    }
```

Comment:
Do you need this with the deprecation flags set on the ParseField? I think so, and it will be logged like `Deprecated field [sub_searches] used, this field is unused and will be removed entirely` or something. Maybe we can indicate that its replaced by `retriever` in the original deprecation adjustment on the parse field above.

Reply:
Updated deprecation message to 
```
"Deprecated field [sub_searches] used, this field is unused and will be removed entirely in a future release. Please consider using the appropriate [retriever] to combine multiple searches."
```